components/Document/pages/DocumentReplicationMarketPage.tsx

```diff
@@ -0,0 +1,351 @@
+import { NextPage } from "next";
```

Comment:
Is this page needed? We may be able to host the page by itself in the `pages/paper/[documentId]/[documentSlug]/[tabName]` 

That would keep our pages consistent

Reply:
I tried creating a separate page at `pages/paper/[documentId]/[documentSlug]/replicability.tsx` but couldn't get it to work — something about the `tabName` param or how the page was structured was interfering with some logic I think.

It's my first time working with Next so wasn't able to figure out the mechanics of why.

I can try to refactor if we think that's the best move, but based off this thread it seems potentially okay to leave as-is?

Reply:
The way the routes work here is that [tabName] is a dynamic parameter. IMO you should put all of your Replication code into a component and then render it here based on the tabName, not create a separate page for it.

Reply:
So you could have an if statement, and do like this: 

```
import { GetStaticPaths, GetStaticProps, NextPage } from "next";
import DocumentCommentsPage from "~/components/Document/pages/DocumentCommentsPage";
import sharedGetStaticProps from "~/components/Document/lib/sharedGetStaticProps";

interface Args {
  documentData?: any;
  commentData?: any;
  errorCode?: number;
  tabName: string;
  metadata?: any;
}

const TabPage: NextPage<Args> = ({
  documentData,
  commentData,
  tabName,
  errorCode,
  metadata,
}) => {
  if (tabName === 'replicability') {
    return (<PaperReplication />)
  }
  return (
    <DocumentCommentsPage
      documentData={documentData}
      documentType="paper"
      metadata={metadata}
      tabName={tabName}
      commentData={commentData}
      errorCode={errorCode}
    />
  );
};

export const getStaticProps: GetStaticProps = async (ctx) => {
  return sharedGetStaticProps({ ctx, documentType: "paper" });
};

export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: [
      //   {
      //   params: {
      //     documentType: 'paper',
      //     documentId: '1276082',
      //     documentSlug: 'boundary-vector-cells-in-the-goldfish-central-telencephalon-encode-spatial-information',
      //     tabName: 'conversation',
      //   },
      // },
    ],
    fallback: true,
  };
};

export default TabPage;
```

Reply:
Isn't that what I did? Ref: https://github.com/ResearchHub/researchhub-web/pull/1586/files#diff-bc02f4ca9eeb4dfc8a2faac8b55e35515b8b25e298300833319d6709f2cfcd1dR20